dashboard/app/controllers/home_controller.rb

```diff
@@ -35,6 +35,8 @@ def index
     if current_user
       @gallery_activities =
         current_user.gallery_activities.order(id: :desc).page(params[:page]).per(GALLERY_PER_PAGE)
+      @request_ip = request.remote_ip
+      @force_race_interstitial = params[:forceRaceInterstitial]
```

Comment:
hoc_detect_country uses request.ip instead of request.remote_ip, but from what I'm seeing online, remote_ip should be better about guessing the actual client IP (as opposed to getting confused by proxies and whatnot)